service/routers/users.js

```diff
@@ -0,0 +1,78 @@
+import { Router } from 'express'
+import { User } from '../models/User.js'
+
+const usersRouter = Router()
+
+function sanitizeUsers(users) {
+    const sanitizedUsers = users.map((user) => ({
+        id: user._id,
+        firstName: user.firstName,
+        lastName: user.lastName,
+        email: user.email,
+        decks: user.decks,
+        active: user.active,
+        role: user.role
+    }))
+    return sanitizedUsers
+}
+
+const getUsers = async (req, res) => {
+  const { userId } = req.user
+  const requestor = await User.findById(userId)
+  if (requestor.role === 'admin' || requestor.role === 'superuser') {
+    const users = await User.find({})
+    res.send(sanitizeUsers(users)) 
+  } else {
+    res.status(403).send('Forbidden')
+  }
+}
+
+const getUsersById = async (req, res) => {
+  const { userId } = req.user   
+  const requestor = await User.findById(userId)  
+  if (requestor.role === 'admin' || requestor.role === 'superuser') {
```

Comment:
You're missing a use case here. The 'user' role should be allowed to access its own user profile